### PR TITLE
Render MathML directly

### PR DESCRIPTION
**Pull request by ronkok**
This PR addresses #1928 by ~~exposing two new functions, `renderMathML()`, and `renderMathMLToString()`~~ adding a rendering option that allows a web site administrator to output KaTeX in HTML only, in MathML only, or in both HTML and MathML (default).  

You can review its effectiveness by using Firefox to view http://webmathcomparison.net/. The KaTeX MathML column on that page now is driven by a build from this branch. I'll be happy to add other tests that might help.

### CODE REVIEW

File: katex.js [hunk 1]
```diff
@@ -151,6 +180,19 @@ export default {
      * for sending to the client.
      */
     renderToString,
+    /**
+     * KaTeX error, usually during parsing.
+     */
```
Comment (kevinbarabash) [inline katex.js]: This comment exists below.  This copy should be removed.

File: src/buildTree.js [hunk 2]
```diff
@@ -57,4 +57,18 @@ export const buildHTMLTree = function(
...
+    tree: AnyParseNode[],
+    expression: string,
+    settings: Settings,
+): DomSpan {
+    const options = optionsFromSettings(settings);
+    const mathMLNode = buildMathML(tree, expression, options);
+    // Flow doesn't think a DomSpan has children.
+    // $FlowFixMe
+    const node = (buildCommon.makeSpan([], [mathMLNode])).children[0].children[0];
+    node.setAttribute("xmlns", "http://www.w3.org/1998/Math/MathML");
```
Comment (kevinbarabash) [inline src/buildTree.js]: Does `buildMathML` not include the namespace?  If not, it probably should and then we can remove this line here.

Comment (ronkok) [reply src/buildTree.js]: > Does `buildMathML` not include the namespace?

Correct. It does not. I'll put the namespace there as you suggest.

Comment (ronkok) [reply src/buildTree.js]: Done.

**Comment by ronkok** [comment]
Control of the output is now moved from the main KaTeX API to the KaTeX rendering options. This should be good to go.

**Pull request merged**